File: exercises/week02/part04/java/src/Trainee.java
package src;

public class Trainee extends Student {
  private String job;
  private float salary;

  public Trainee() {
    System.out.println("Trainee class constructor.");
  }

  public Trainee(String name, String studentId, String job, float salary) {
    this.setName(name);
    this.setStudentId(studentId);
    this.job = job;
    this.salary = salary;

    System.out.println("Trainee class parameterized constructor.");
  }

  //getters

  public String getJob() { return this.job; }
  public float getSalary() { return this.salary; }

  //setters

  public void setJob(String job) {
    this.job = job;
  }

  public void setSalary(float salary) {
    this.salary = salary;
  }

  //public methods

  public void print() {
    super.print();
    var info = String.format("Job: %s\nSalary: %.2f",
      this.job,
      this.salary
    );
    System.out.println(info);
  }
}

File: exercises/week02/part01/src/trainee/trainee.cpp
#include "trainee.h"

Trainee::Trainee(void) {}

Trainee::Trainee(const std::string& name, const std::string& student_id, const std::string& job, const float& salary)
  : Student(name, student_id), _job(job), _salary(salary) {}

Trainee::~Trainee() {}

//getters

std::string Trainee::job(void) const { return _job; } 
float Trainee::salary(void) const { return _salary; }

//setters

void Trainee::set_job(const std::string& job) {
  _job = job;
}

void Trainee::set_salary(const float& salary) {
  _salary = salary;
}

//public methods

void Trainee::print(void) const {
  Student::print();
  printf("Job: %s\nSalary: %.2f\n", _job.c_str(), _salary);
}